pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -155,27 +155,37 @@ public void updateDefaultColumns()
    *
    * @return Action Map for each column.
    */
-  private Map<String, DefaultColumnAction> computeDefaultColumnActionMap() {
+  public static Map<String, DefaultColumnAction> computeDefaultColumnActionMap(Schema schema,
```

Comment:
Not sure if we still need to modify this class. It should remain the same for the completed segments, right?

Reply:
Removed unused code for the consuming segment. I kept some changes for the unit tests that I added for the committed segment